### ISSUE TITLE
EXCEPTION: different prefix: "C:/" and "D:/...."

### ISSUE THREAD

**Issue by alexmarincu** [ThrowTheSwitch/Ceedling#944]
Hi, I'm using the latest pre-release 1.0.0 version on a Windows machine and I have the following issue when running `ceedling test:all` command:

`EXCEPTION: different prefix: "C:/" and "D:/...."`

This happens when I try to include a project dependency using absolute path. Here is a snippet from the project.yml file:

```
:paths:
    :test:
        - +:test/**
    :source:
        - src/**
        - deps/**
    :include:
        - src/**
        - deps/**
        - C:/Keil_v5/ARM/Packs/ARM/CMSIS/6.1.0/CMSIS/Core/Include/**
```

My project is located somewhere on D:/ drive, but it has dependencies on some Keil uVision packs. Keil uVision and the packs are located on the C:/ drive. I need to include them from their original location (C:/) without the need to create symlinks.
Am I doing something wrong or this configuration is not supported?

**Comment by mvandervoord** [comment]
Hi. That's an interesting problem. Can you rerun the experiment as `ceedling -v=4 test:all` (aka `--verbosity=obnoxious`). It would be helpful to get a more complete stack trace to see what part of the system is throwing the error. :)

**Comment by alexmarincu** [comment]
It seems like you are right to suspect ruby's `relative_path_from` function:

```
❯ ceedling --debug test:all
Loaded project configuration from working directory.
 > Using: D:/Data/projects/my-project/project.yml
 > Working directory: D:/Data/projects/my-project
 > Set which Ceedling from config :project >> :which_ceedling => gem
 > Launching Ceedling from C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/

Application & Build Frameworks
   Ceedling => 1.0.0-cb01ae0
      CMock => 2.6.0
      Unity => 2.6.0
 CException => 1.3.4

Validating configuration contains minimum required sections...

Project Configuration Handling
------------------------------
Processing Unity configuration...
Processing CMock configuration...

Plugin Handling
---------------
Discovering all plugins...
 > Rake plugins: gcov, module_generator
 > Programmatic plugins: gcov, module_generator, report_tests_pretty_stdout, report_tests_log_factory
 > Config plugins: module_generator, report_tests_pretty_stdout
Merging configuration from plugin module_generator...
{:module_generator=>{:project_root=>"./", :naming=>:snake, :boilerplates=>{:src=>"", :inc=>"", :tst=>""}}}
Merging configuration from plugin report_tests_pretty_stdout...
{:plugins=>{:display_raw_test_results=>false}}

Assembling Default Settings
---------------------------
Collecting default tool configurations...
Collecting CMock defaults...
Collecting Plugin YAML defaults...
 - gcov >> {:gcov=>{:summaries=>true, :report_task=>false, :utilities=>["gcovr"], :reports=>[], :gcovr=>{:report_root=>".", :merge_mode_function=>"merge-use-line-max"}, :report_generator=>{:verbosity=>"Warning", :collection_paths_source=>[], :custom_args=>[], :gcov_exclude=>[]}}}
 - report_tests_log_factory >> {:report_tests_log_factory=>{:reports=>[]}}
Collecting Plugin Ruby hash defaults...
 - gcov >> {:tools=>{:gcov_compiler=>{:executable=>"gcc.exe", :name=>"default_gcov_compiler", :optional=>false, :arguments=>["-g", "-fprofile-arcs", "-ftest-coverage", "-I\"${5}\"", "-D\"${6}\"", "-DGCOV_COMPILER", "-DCODE_COVERAGE", "-c \"${1}\"", "-o \"${2}\"", "-MMD", "-MF \"${4}\""]}, :gcov_linker=>{:executable=>"gcc.exe", :name=>"default_gcov_linker", :optional=>false, :arguments=>["-g", "-fprofile-arcs", "-ftest-coverage", "${1}", "${5}", "-o \"${2}\"", "${4}"]}, :gcov_fixture=>{:executable=>"${1}", :name=>"default_gcov_fixture", :optional=>false, :arguments=>[]}, :gcov_summary=>{:executable=>"gcov.exe", :name=>"default_gcov_summary", :optional=>true, :arguments=>["-n", "-p", "-b", "-o \"${2}\"", "\"${1}\""]}, :gcov_report=>{:executable=>"gcov.exe", :name=>"default_gcov_report", :optional=>true, :arguments=>["-b", "-c", "-r", "-x", "${1}"]}, :gcov_gcovr_report=>{:executable=>"gcovr", :name=>"default_gcov_gcovr_report", :optional=>true, :arguments=>["${1}"]}, :gcov_reportgenerator_report=>{:executable=>"reportgenerator.exe", :name=>"default_gcov_reportgenerator_report", :optional=>true, :arguments=>["${1}"]}}}
Populating project configuration with collected default values...

Completing Project Configuration
--------------------------------
Populating test runner generation settings...
Unity configuration >> {:defines=>["UNITY_EXCLUDE_FLOAT"], :vendor_path=>"C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/vendor", :use_param_tests=>false}      
CMock configuration >> {:plugins=>[:ignore, :callback], :verbosity=>2, :when_no_prototypes=>:warn, :mock_prefix=>"mock_", :memcmp_if_unknown=>true, :when_ptr=>:compare_data, :weak=>"", :enforce_strict_ordering=>true, :fail_on_unexpected_calls=>false, :callback_include_count=>true, :callback_after_arg_check=>false, :exclude_setjmp_h=>false, :vendor_path=>"C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/vendor", :includes=>[], :unity_helper_path=>[], :defines=>[], :mock_suffix=>"", :mock_path=>"build/test/mocks"}
Test Runner configuration >> {:cmdline_args=>true, :includes=>[], :defines=>["UNITY_EXCLUDE_FLOAT"], :file_suffix=>"_runner", :mock_prefix=>"mock_", :mock_suffix=>"", :enforce_strict_ordering=>true, :use_param_tests=>false}
CException configuration >> {:vendor_path=>"C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/vendor", :defines=>[]}
Processing environment variables...
Processing path entries and expanding any string replacements...
Expanding any string replacements in :flags entries...
Expanding any string replacements in :defines entries...
Standardizing all paths...
Populating tool definition settings and expanding any string replacements...
Processing tool definition shortcuts...

Validating final project configuration...
Build paths:
 - build/test/mocks
 - build/artifacts
 - build/test
 - build/vendor
 - build/artifacts/test
 - build/test/runners
 - build/test/results
 - build/test/out
 - build/test/cache
 - build/test/dependencies
 - build/vendor/unity/src
 - build/vendor/cmock/src
 - build/logs
 - build/test/preprocess/includes
 - build/test/preprocess/files

EXCEPTION: different prefix: "C:/" and "D:/Data/projects/my-project"
Backtrace ==>
C:/prog/scoop/apps/ruby/3.3.5-1/lib/ruby/3.3.0/pathname.rb:534:in `relative_path_from': different prefix: "C:/" and "D:/Data/projects/my-project" (ArgumentError)
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/file_path_collection_utils.rb:84:in `block in collect_paths'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/file_path_collection_utils.rb:82:in `map!'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/file_path_collection_utils.rb:82:in `collect_paths'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/configurator_builder.rb:272:in `block in expand_all_path_globs'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/configurator_builder.rb:270:in `each'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/configurator_builder.rb:270:in `expand_all_path_globs'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/configurator_setup.rb:94:in `build_project_collections'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/configurator.rb:622:in `build'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/setupinator.rb:165:in `do_setup'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/lib/ceedling/rakefile.rb:65:in `<top (required)>'
        <internal:C:/prog/scoop/apps/ruby/3.3.5-1/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
        <internal:C:/prog/scoop/apps/ruby/3.3.5-1/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/bin/cli_helper.rb:133:in `load_ceedling'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/bin/cli_handler.rb:223:in `build'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/bin/cli.rb:346:in `build'
        C:/prog/scoop/persist/ruby/gems/gems/thor-1.3.1/lib/thor/command.rb:28:in `run'
        C:/prog/scoop/persist/ruby/gems/gems/thor-1.3.1/lib/thor/invocation.rb:127:in `invoke_command'
        C:/prog/scoop/persist/ruby/gems/gems/thor-1.3.1/lib/thor.rb:527:in `dispatch'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/bin/cli.rb:93:in `start'
        C:/prog/scoop/persist/ruby/gems/gems/ceedling-1.0.0/bin/ceedling:125:in `<top (required)>'
        C:/prog/scoop/apps/ruby/current/gems/bin/ceedling:32:in `load'
        C:/prog/scoop/apps/ruby/current/gems/bin/ceedling:32:in `<main>'
```

**Comment by mvandervoord** [comment]
Thanks. I'll see if I can tackle that problem quickly! :)

**Comment by alexmarincu** [comment]
Thank you too for the quick follow-up. I appreciate your effort and I am grateful for this fantastic tool.

**Comment by mvandervoord** [comment]
No worries. Sometimes we get lucky with the timing and I'm able to work on something immediately. :)

I believe I have this issue fixed. When you have time, can you grab prerelease  [1.0.0-8ce41a0](https://github.com/ThrowTheSwitch/Ceedling/releases/tag/1.0.0-8ce41a0) and give it a try?

**Comment by alexmarincu** [comment]
It's fixed in prerelease [1.0.0-8ce41a0](https://github.com/ThrowTheSwitch/Ceedling/releases/tag/1.0.0-8ce41a0). Thank you :)
